app/controllers/admin/resources_controller.rb

```diff
@@ -23,12 +24,19 @@ def create
     end
 
     def update
-      if @resource.update_attributes(resource_params)
-        redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
-                    notice: 'Resource successfully updated.'
-      else
-        flash.now[:error] = "Resource update failed: #{@resource.errors.full_messages.join('. ')}."
-        render :edit
+      successful_update = check_successful_update params[:increment_used_resource_flag]
+      respond_to do |format|
+        if successful_update
+          format.html do
+            redirect_to admin_conference_resources_path(conference_id: @conference.short_title),
+                        notice: 'Resource successfully updated.'
+          end
+          flash.now[:notice] = get_successful_update_flash params[:increment_used_resource_flag]
```

Comment:
Try coming back to your code 1-2 days after you have submitted your PR. Then go to the changed files and read your changes. Does it help you to jump from place to place to find what the controller does?